### PR TITLE
Pre-release v2.0.0

### PR DESCRIPTION
**Pull request by jeipollack**
Refactored wf-psf code including:

- New User Interface
- Modular package design
- Unit testing Framework
- Continuous Integration
- Continuous Documentation

New changes to config parameter handling to allow for a bit more flexibility for running WaveDiff under different configurations.
Clean up to plot file naming command & missing doc strings

### CODE REVIEW

File: .github/workflows/cd.yml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: CD
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - feature_documentation
+      - dummy_main
```
Comment (sfarrens) [inline .github/workflows/cd.yml]: Update to `main`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: CI
+
+on:
+  pull_request:
+    branches:
+      - dummy_main
```
Comment (sfarrens) [inline .github/workflows/ci.yml]: Update to `main`

File: debug/debug_tf_model.py
Comment (sfarrens) [inline debug/debug_tf_model.py:1]: What does this module do?

File: docs/source/conf.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+
+sys.path.insert(0, os.path.abspath("src/wf_psf"))
+
+# -- Project information -----------------------------------------------------
+# https://www.sphinx-doc.org/en/master/usage/configuration.html#project-information
+
+project = "wf-psf"
+copyright = "2023, CosmoStat"
+author = "CosmoStat"
+release = "1.0.0"
```
Comment (sfarrens) [inline docs/source/conf.py]: Documentation version does not match that in the metadata. This should be `2.0.0`.

**Review by sfarrens — CHANGES_REQUESTED** [review]
Hi @jeipollack, just a few minor threads to resolve. I don't anticipate any further issues after you merge #85, so once that is done and you have resolve these threads, I believe we can merge this PR too.

File: .github/workflows/cd.yml (hunk 1 above)
Comment (jeipollack) [reply .github/workflows/cd.yml]: done.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (jeipollack) [reply .github/workflows/ci.yml]: done

File: debug/debug_tf_model.py
Comment (jeipollack) [reply debug/debug_tf_model.py:1]: removed the entire debug folder since it is not used.

File: docs/source/conf.py (hunk 3 above)
Comment (jeipollack) [reply docs/source/conf.py]: updated

File: .github/workflows/cd.yml (hunk 1 above)
Comment (sfarrens) [reply .github/workflows/cd.yml]: I think you can remove `feature_documentation`. I don't think that branch exists anymore.

File: .github/workflows/cd.yml [hunk 4]
```diff
@@ -0,0 +1,44 @@
+# This workflow will install Python dependencies, run tests and lint with a single version of Python
+# For more information see: https://docs.github.com/en/actions/automating-builds-and-tests/building-and-testing-python
+
+name: CD
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - feature_documentation
```
Comment (sfarrens) [inline .github/workflows/cd.yml]:
```suggestion
```

Comment (jeipollack) [reply .github/workflows/cd.yml]: oh I thought this was for the CD GitHub-pages.  but now I remember the branch is called gh-pages

Comment (jeipollack) [reply .github/workflows/cd.yml]: I removed the extra branch and the `workflow_dispatch` trigger

**Review by sfarrens — APPROVED** [review]

**Pull request merged**